cluster/plan.go

```diff
@@ -509,8 +509,8 @@ func (c *Cluster) BuildKubeletProcess(host *hosts.Host, serviceOptions v3.Kubern
 		if err != nil {
 			logrus.Debugf("Error while parsing cluster version: %s", err)
 		}
-		// cri-dockerd must be enabled if the cluster version is 1.24 and higher
-		if parsedRangeAtLeast124(parsedVersion) {
+
+		if useNewCRIEndpoint(c.Version, parsedVersion) {
 			CommandArgs["container-runtime-endpoint"] = "unix:///var/run/cri-dockerd.sock"
 			Binds = []string{fmt.Sprintf("%s:/var/lib/cri-dockerd:z", path.Join(host.PrefixPath, "/var/lib/cri-dockerd"))}
```

Comment:
So the fix for the following issue will also be available on the latest 1.21 1.22 1.23, right? Do we want to do it? My concern is it will require extra validations from QA. 
https://github.com/rancher/rke/issues/2993 ( its PR https://github.com/rancher/rke/pull/3001)